aspnetcore/security/authorization/roles/samples/3_0/Startup.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Identity.UI;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using WebApplication13.Data;
+
+namespace WebApplication13
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        #region snippet
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<ApplicationDbContext>(options =>
+                options.UseSqlServer(
+					Configuration.GetConnectionString("DefaultConnection")));
+            services.AddIdentity<IdentityUser, IdentityRole>()
+                .AddRoles<IdentityRole>()
```

Comment:
I don't think `AddRoles` is needed here as this is already handled using `AddIdentity`, but I do see that the section is about adding this call. Should this just use `AddDefaultIdentity` too, as in the 2.2 version?:

```csharp
services.AddDefaultIdentity<IdentityUser>()
    .AddRoles<IdentityRole>()
    .AddEntityFrameworkStores<ApplicationDbContext>();
```

Reply:
`AddIdentityCore` perhaps?  altho then that requires calling add authentication.  Or maybe remove the call to `AddDefaultUI` instead

Reply:
I updated this to the fix from @serpent5 , Is this ok?